src/Provider/TMSProvider.js

```diff
@@ -23,32 +24,50 @@ function preprocessDataLayer(layer) {
             max: 18,
         };
     }
+    layer.fx = layer.fx || 0.0;
+}
+
+function getVectorTile(tile, coords, layer) {
+    const url = URLBuilder.xyz(coords, layer);
+
+    if (layer.type == 'color') {
+        return VectorTileHelper.getVectorTileTextureByUrl(url, tile, layer, coords);
+    }
 }
 
 function executeCommand(command) {
     const layer = command.layer;
     const tile = command.requester;
 
     const promises = [];
+    const supportedFormats = {
+        'application/x-protobuf;type=mapbox-vector': getVectorTile.bind(this),
+    };
+    const func = supportedFormats[layer.format];
+
     for (const coordTMS of tile.getCoordsForLayer(layer)) {
         const coordTMSParent = (command.targetLevel < coordTMS.zoom) ?
             OGCWebServiceHelper.WMTS_WGS84Parent(coordTMS, command.targetLevel) :
             undefined;
+        if (func) {
+            promises.push(func(tile, coordTMSParent || coordTMS, layer));
+        } else {
+            const urld = URLBuilder.xyz(coordTMSParent || coordTMS, layer);
 
-        const urld = URLBuilder.xyz(coordTMSParent || coordTMS, layer);
-
-        promises.push(OGCWebServiceHelper.getColorTextureByUrl(urld, layer.networkOptions).then((texture) => {
-            const result = {};
-            result.texture = texture;
-            result.texture.coords = coordTMSParent || coordTMS;
-            result.pitch = coordTMSParent ?
-                coordTMS.offsetToParent(coordTMSParent) :
-                new THREE.Vector4(0, 0, 1, 1);
-            if (layer.transparent) {
-                texture.premultiplyAlpha = true;
-            }
-            return result;
-        }));
+            promises.push(OGCWebServiceHelper.getColorTextureByUrl(urld, layer.networkOptions).then((texture) => {
+                const result = {};
+                const pitch = coordTMSParent ?
+                    coordTMS.offsetToParent(coordTMSParent) :
+                    new THREE.Vector4(0, 0, 1, 1);
+                result.texture = texture;
+                result.texture.coords = coordTMSParent || coordTMS;
+                result.pitch = pitch;
+                if (layer.transparent) {
+                    texture.premultiplyAlpha = true;
+                }
+                return result;
+            }));
+        }
```

Comment:
now that you use `coordTMSParent` if `func` is defined, dont you need to introduce this postprocessing (most notably `pitch`) in all cases ?